atc/db/container.go

```diff
@@ -385,26 +385,44 @@ func (container *failedContainer) Handle() string              { return containe
 func (container *failedContainer) WorkerName() string          { return container.workerName }
 func (container *failedContainer) Metadata() ContainerMetadata { return container.metadata }
 
-func (container *failedContainer) Destroy() (bool, error) {
-	rows, err := psql.Delete("containers").
-		Where(sq.Eq{
-			"id":    container.id,
-			"state": atc.ContainerStateFailed,
+func (container *failedContainer) Destroy() (DestroyingContainer, error) {
+	result, err := psql.Update("containers").
+		Set("state", atc.ContainerStateDestroying).
+		Where(sq.And{
+			sq.Eq{"id": container.id},
+			sq.Or{
+				sq.Eq{"state": string(atc.ContainerStateDestroying)},
```

Comment:
I might just be missing some context but I'm just curious, why are allowing destroying and created containers to be updated within a failed container object?

Reply:
copy pasta